File: korecomm-go/pkg/comm/ingress_message.go
package comm

// Originator contains originating information for an `IngressMessage`. It
// contains the "identity" of the person/system that triggered the incoming
// message, in addition to the adapter name that procuded the event.
type Originator struct {
	Identity    string
	AdapterName string
}

// IngressMessage is the structured, parsed message representing an incoming
// command to be processed. An `IngressMessage` has been determined by the
// system to have content containing a command.
type IngressMessage struct {
	Content    string
	Originator Originator
}

// RawIngressMessage - Raw, unprocessed message passed from the adapter to the
// engine. Has not yet been parsed to determine if the message is a cmd or not.
type RawIngressMessage struct {
	Identity   string
	RawContent string
}

File: korecomm-ruby/lib/kore/comm/ingress_message.rb
module Kore
  module Comm
    class IngressMessage
      attr_accessor :originator, :plugin, :cmd, :content, :raw
      def initialize(o)
        @originator = o.fetch(:originator, '')
        @plugin = o.fetch(:plugin, '')
        @content = o.fetch(:content, '')
      end

      def to_s
        "[#{@originator}] [#{@plugin}] [#{@content}]"
      end
    end
  end
end
